dumux-precice/couplingadapter.hh

```diff
@@ -287,97 +195,67 @@ public:
      */
     size_t getNumberOfVertices();
     /*!
-     * @brief Gets value of a scalar quantity.
+     * @brief Reads full block of data from preCICE.
      *
-     * @param[in] dataID Identifier of the quantity.
-     * @param[in] faceID Identifier of the face according to DuMuX' numbering.
-     * @return double Value of scalar quantity.
+     * @param[in] meshName Name of the mesh.
+     * @param[in] dataName Name of the data.
+     * @param[in] relativeReadTime The relative time tagged to the data to be read.
+     */
+    void readQuantityFromOtherSolver(const precice::string_view &meshName,
+                                     const precice::string_view &dataName,
+                                     double relativeReadTime);
+    /*!
+     * @brief Writes full block of data to preCICE.
+     *
+     * @param[in] meshName Name of the mesh.
+     * @param[in] dataName Name of the data.
      */
-    double getScalarQuantityOnFace(const size_t dataID, const int faceID) const;
-    // /*!
-    //  * @brief Gets value of a vector quantity.
-    //  *
-    //  * @param[in] dataID Identifier of the quantity.
-    //  * @param[in] faceID Identifier of the face according to DuMuX' numbering.
-    //  * @return std::vector<double> Value of vector quantity.
-    //  */
-    // std::vector<double> getVectorQuantityOnFace(const size_t dataID, const int faceID) const;
-    // std::vector<double> getVectorQuantity(const size_t dataID) const;
+    void writeQuantityToOtherSolver(const precice::string_view &meshName,
```

Comment:
Why are we changing the name `getScalarQuantityOnFace` with `writeQuantityToOtherSolver`?